src/intro.md

```diff
@@ -1,35 +1,62 @@
-  changes to web servers.
-- The Notary and the Verifier can be the same entity. That means if you as a
-  Verifier do not want to trust some Notary server, you can run one yourself.
+## Data Provenance without Compromising Privacy, That is Why!
 
+The Internet currently lacks effective, privacy-preserving **Data Provenance**. [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security), also known as the "s" in "https" 🔐 to the general public, ensures that data can be securely communicated between a server and a user. But how can this user share this data with another user or server without compromising security, privacy, and control?
+
+Enter TLSNotary: a protocol enabling users to export data securely from any website. This data can be selectively shared, allowing others to cryptographically verify its authenticity and integrity. By leveraging Zero Knowledge Proof (ZKP) technology, TLSNotary guarantees authentic, selective data disclosure.
+
+True data provenance will make data truly portable and allow users to share it with others as they see fit.
+
+## How Does the TLSNotary Protocol Work?
+
+The TLSNotary protocol consists of 4 steps:
+1. The `User` **requests** the data from the server privately and securely.
+2. The Notary **checks** for data tampering and **signs** the transcript.
+3. The `User` **selectively discloses** the data to the `Verifier`.
+4. The `Verifier` **verifies** the data.
+
+![](./png-diagrams/overview3.png)
+
+### ① Multi-party TLS Request
+
+TLSNotary works by adding a third party, the `Notary`, to the usual TLS connection between the `User` and a Server. This `Notary` is **not an intermediary**. Instead, the `Notary` participates in a **secure multi-party computation** (MPC) to jointly operate the TLS connection without ever seeing the data in plain text; the `Notary` only sees encrypted data. Given that the `Notary` only sees the ephemeral keys of the Server, the `Notary` does not know which Server the `User` is communicating with. The TLSNotary protocol is transparent to the Server. From the Server's perspective, the TLS connection is the same as all other connections. Therefore, no modifications to the TLS protocol are needed.
+
+<!-- - Transport Layer Security (TLS)
+    - Encryption: hides data from third parties
+    - Authentication: ensures that the parties exchanging information are who they claim to be
+    - Integrity: verifies that data has not been forged or tampered with -->
+### ② Notarization
+
+The `User` forwards the encrypted TLS traffic to the `Notary`, which validates its integrity and then notarizes the entire TLS session by signing its transcript.
+
+### ③ Selective Disclosure
+
+The TLSNotary protocol offers a way for the `User` to selectively prove the authenticity of arbitrary portions of the transcript to a Verifier. Because of the way that the `Notary` signs the transcript, the `User` can prove the authenticity of the transcript to any third-party `Verifier` using Zero Knowledge Proofs.
```

Comment:
My understanding here is a little bit different.
Selective disclosure means that the user can redact parts of the transcript when showing it to the verifier. There are no ZK proofs involved.

ZK proofs come into play when you want to prove something in ZK about redacted parts, like e.g. "the age field is redacted, but here is the ZK proof that it is over 18".